php/Block.php

```diff
@@ -63,66 +63,179 @@ public function register_block() {
 	 * @return string The markup of the block.
 	 */
 	public function render_callback( $attributes, $content, $block ) {
-		$post_types = get_post_types(  [ 'public' => true ] );
-		$class_name = $attributes['className'];
-		ob_start();
 
+		// Store this in a local var for caching since we'll reference it a few
+		// times in this method.
+		$current_post_id = get_the_ID();
```

Comment:
Typically I wouldn't actually cache this locally assuming that `get_the_ID()` is already going to be well cached and easily readable, but it's a pattern I've seen others use.